utils/src/tree/v2/ActivityChildItem.ts

```diff
@@ -0,0 +1,36 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { TreeItem, TreeItemCollapsibleState } from "vscode";
+import { TreeItemIconPath } from "../../..";
+import * as types from '../../../index';
+
+export class ActivityChildItem implements types.ActivityChildItemBase {
+    readonly id: string;
+    label: string;
+    contextValue: string;
+    description?: string;
+    iconPath?: TreeItemIconPath;
+    initialCollapsibleState?: TreeItemCollapsibleState;
+
+    constructor(options: types.ActivityChildItemOptions) {
+        this.id = options.id;
+        this.label = options.label;
+        this.contextValue = options.contextValue;
+        this.description = options.description;
+        this.iconPath = options.iconPath;
+        this.initialCollapsibleState = options.initialCollapsibleState;
```

Comment:
Maybe we should include the ActivityStatus (it's an enum that has failed/success/running/etc.) on this as well?

That way you can just check its value instead of looking at the context value and doing the regex test on it as well.

Reply:
Yeah that could be nifty.  For this scenario, would you want the activity status to be a part of `ActivityChildItemBase` as well then?

Reply:
Yeah, I think that would make sense. It can be optional and default to "NotStarted" if not found, maybe?

Reply:
This one?

![image](https://github.com/user-attachments/assets/48fb8b69-c1f0-4de5-8018-6a1a0a0e197b)

I was thinking some more about this, and I don't know if `NotStarted` would apply.  Execute steps only provide items once they've started running, so anything `NotStarted` just wouldn't show up I think.  Also, not all items we discussed using will necessarily go through this full lifecycle.  It makes sense for the `ActivityItem`, just not sure if it's the best representation for `ActivityChildItem`.

Is something like `activityType` more helpful?  Example:

1. 'success'
2. 'fail'
3. 'progress'
4. 'info'
5. 'error'
6. Anything else?

I assume you can infer the `activityStatus` (when it applies) based on this type?  Let me know what you guys are thinking

Reply:
Yeah, I get what you mean. This feels fine to me.

I'm not sure I 100% understand the distinction between fail and error.

Reply:
Oh yeah let me provide a couple images that will better help illustrate the difference by what I mean there

Reply:
![image](https://github.com/user-attachments/assets/209961f5-2115-469b-8924-20751a881702)
`Create Network Security Group....` is a fail item indicating the step that failed
`Network security group cannot be created...` is an error item indicating the error message